src/components/website-management/website-form-components/customization/customization-components/CookieBanner.js

```diff
@@ -0,0 +1,331 @@
+  const [error, setError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState("");
+  const [showDialog, setShowDialog] = useState(false);
+
+  const handleCloseClick = () => {
+    setShowDialog(false);
+  };
+
+  const handleCustomizeClick = () => {
+    setShowDialog((prevShowDialog) => !prevShowDialog);
+  };
+
+  const loadTemplates = async () => {
+    try {
+      const websiteID = passedId;
+      const result = await axios.get(
+          `http://localhost:8080/api/v1/templates/${websiteID}`
+      );
+      settemplates(result.data);
+
+      if (Object.keys(selectedTemplate).length !== 0) {
+        const template = result.data.find(
+            (t) => t.templateName === selectedTemplate?.templateName
+        );
+        setSelectedTemplate(template);
+      } else {
+        const template = result.data[0];
+        setSelectedTemplate(template);
+      }
+    }
+    catch (error) {
+      console.error("Error loading banners: ", error);
+    }
+  };
+
+  const laodBanner = async () => {
+
+    try {
+      const websiteId = passedId;
+      const response = await axios.get(
+        `http://localhost:8080/api/v1/banners/${websiteId}`
+      );
+      const result = response.data;
+      setBanner(result);
+
+      const { bannerPosition, bannerColor, bannerAlignment,bannerTextColor } = result;
+      setPositionValue(bannerPosition);
+      handleChangeTop(bannerPosition);
+      handleChangeHorizontal(bannerAlignment);
+      handleChangeColor(bannerColor);
+      handleChangeTextColor(bannerTextColor);
+
+    } catch (error) {
+      console.error("Error loading banners:", error);
+    }
+  };
+
+  useEffect(() => {
+    loadTemplates();
+    laodBanner();
+  }, []);
+
+  const templateValues =
+    templates?.length > 0
+      ? templates.map((template) => ({
+          id: template.templateId,
+          value: template.templateName,
+        }))
+      : [];
+
+
+  const choosenTemplate = (e) => {
+    const selectedRegulation = e.target.value;
+    let chosenCont = null;
+
+    for (let i = 0; i < templates.length; i++) {
+      if (templates[i].templateName === selectedRegulation) {
+        chosenCont = templates[i].templateContent;
+        break;
+      }
+    }
+
+    const template = templates.find(
+      (t) => t.templateName === e.target.value
+    );
+    setSelectedTemplate(template);
+    setChosenCont(chosenCont);
+  };
+
+  function handleChangeTop(value) {
+    if (value === "Top") {
+      setTop(10);
+      setPositionValue(value);
+    }
+    if (value === "Bottom") {
+      setTop(280);
+      setPositionValue(value);
+    }
+  }
+
+  function handleChangeHorizontal(value) {
+    if (value === "Left") {
+      setLeft(10);
+      setAlignmentValue(value);
+    }
+    if (value === "Center") {
+      setLeft(240);
+      setAlignmentValue(value);
+    }
+    if (value === "Right") {
+      setLeft(475);
+      setAlignmentValue(value);
+    }
```

Comment:
Let's define an enum for these string literals and use that here, instead of just using plain strings.

Ref: https://masteringjs.io/tutorials/fundamentals/enum

Reply:
noted